### PR TITLE
Add data protection service pw 2888

### PR DESCRIPTION
**Pull request by zurmalik**
**Description**
Add data protection subject erasure

**Tested scenarios**
From postman a "PAYMENT_NOT_FOUND" scenario was executed with result HTTP status 200 

**Fixed issue**:  pw-2888

### CODE REVIEW

File: src/main/java/com/adyen/service/DataProtection.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    private RequestSubjectErasure requestSubjectErasure;
+
+    public DataProtection(Client client) {
+        super(client);
+        requestSubjectErasure = new RequestSubjectErasure(this);
+    }
+
+    /**
+     * POST /erase API call
```
Comment (Aleffio) [inline src/main/java/com/adyen/service/DataProtection.java:44]:
```suggestion
     * POST /requestSubjectErasure API call
```

File: src/test/java/com/adyen/DataProtectionServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import com.adyen.service.DataProtection;
+import com.adyen.service.exception.ApiException;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests for /authorise and /authorise3d
```
Comment (Aleffio) [inline src/test/java/com/adyen/DataProtectionServiceTest.java:34]:
```suggestion

```

File: src/main/java/com/adyen/model/SubjectErasureResponse.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+   * (except the first line).
+   */
+  private String toIndentedString(Object o) {
+    if (o == null) {
+      return "null";
+    }
+    return o.toString().replace("\n", "\n    ");
+  }
+
+}
```
Comment (Aleffio) [inline src/main/java/com/adyen/model/SubjectErasureResponse.java:136]: can you please auto-format all your new classes?(option, cmd, l)

File: src/test/java/com/adyen/DataProtectionServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        DataProtection dataProtection = new DataProtection(client);
+        SubjectErasureRequest subjectErasureRequest = createSubjectErasureRequest();
+
+        try {
+            dataProtection.requestSubjectErasure(subjectErasureRequest);
+            fail("Exception expected");
+        } catch (ApiException e) {
+            assertNotNull(e.getError());
+            assertEquals(404, e.getStatusCode());
```
Comment (Aleffio) [inline src/test/java/com/adyen/DataProtectionServiceTest.java:72]: I believe this shouldn't return a 404 if it has a result in it, let's double check

File: src/main/java/com/adyen/model/SubjectErasureResponse.java (hunk 3 above)
Comment (zurmalik) [reply src/main/java/com/adyen/model/SubjectErasureResponse.java:136]: done. pushing change

Comment (Aleffio) [reply src/main/java/com/adyen/model/SubjectErasureResponse.java:136]: SubjectErasureResponse is not formatted yet, can you please format this one too?

File: README.md [hunk 5]
```diff
@@ -39,7 +40,7 @@ Add this dependency to your project's POM:
 <dependency>
   <groupId>com.adyen</groupId>
   <artifactId>adyen-java-api-library</artifactId>
-  <version>9.0.0</version>
+  <version>9.0.1</version>
```
Comment (Aleffio) [inline README.md:43]: we made some breaking changes here: https://github.com/Adyen/adyen-java-api-library/pull/438
so we need to upgrade to a major version, can you please bump it to 10.0.0?

File: src/main/java/com/adyen/model/SubjectErasureResponse.java (hunk 3 above)
Comment (zurmalik) [reply src/main/java/com/adyen/model/SubjectErasureResponse.java:144]: pushed

File: README.md (hunk 5 above)
Comment (zurmalik) [reply README.md]: done. pushed.

**Review by Aleffio — APPROVED** [review]

**Review by andrei-dediu — APPROVED** [review]

**Pull request merged**